src/hotspot/share/gc/shenandoah/shenandoahGeneration.cpp

```diff
@@ -882,43 +882,45 @@ size_t ShenandoahGeneration::increment_affiliated_region_count() {
   // During full gc, multiple GC worker threads may change region affiliations without a lock.  No lock is enforced
   // on read and write of _affiliated_region_count.  At the end of full gc, a single thread overwrites the count with
   // a coherent value.
```

Comment:
Yes It is atomic, the lock/safepoint seems not needed. I'll probably keep the comment  as it is in this PR, since the are called from different places of FullGC and concurrentGC, we can cleanup these methods later I think.

Reply:
ok for now. Will be worth examining the uses from full at some point but this was just a comment so ok for now. May be leave a `TODO` comment to track if you feel like.

```
// TODO: Check and correct comment, if obsolete.
```

Reply:
Thanks, I'll add it if I amend any new change to this PR.